kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/filter/DenyCipherSuiteFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import io.netty.handler.ssl.CipherSuiteFilter;
+import io.netty.util.internal.EmptyArrays;
+
+/**
+ * This class has been built to filter cipher suites based on requirements for Kroxylicious
+ * If no ciphers are declared then the default platform ciphers will be used
+ * If ciphers are declared then they are used instead of the default platform ciphers
+ * Both lists would then have anything removed that wasn't supported or in the deniedCiphers
+ */
+public final class DenyCipherSuiteFilter implements CipherSuiteFilter {
+
+    public static final DenyCipherSuiteFilter INSTANCE = new DenyCipherSuiteFilter();
+
+    private DenyCipherSuiteFilter() {
+    }
+
+    private List<String> deniedCiphers;
+
+    public DenyCipherSuiteFilter deniedCiphers(List<String> deniedCiphers) {
+        this.deniedCiphers = deniedCiphers == null ? new ArrayList<>() : deniedCiphers;
+        return this;
+    }
+
+    @Override
+    public String[] filterCipherSuites(Iterable<String> ciphers, List<String> defaultCiphers,
+                                       Set<String> supportedCiphers) {
+        return ciphers == null ? allowedCiphers(defaultCiphers.size(), defaultCiphers, supportedCiphers)
+                : allowedCiphers(supportedCiphers.size(), ciphers, supportedCiphers);
+    }
+
+    private String[] allowedCiphers(int newListSize, Iterable<String> ciphersToUse, Set<String> supportedCiphers) {
+        List<String> newCiphers = new ArrayList<>(newListSize);
+        for (String cipher : ciphersToUse) {
+            if (cipher == null) {
+                break;
```

Comment:
when would we expect a null value to be contained in the `ciphersToUse`? maybe the `AllowDeny` should disallow null items so that:

```yaml
cipherSuites:
  allow: null
  deny: null
```
is valid, but:
```yaml
cipherSuites:
  allow:
  - null
  - 'cipher'
  deny:
  - null
  - 'cipher2'
```
is invalid and this code should either skip the null or fail with explosion rather than breaking out of the loop.

Reply:
Hi @robobario I'd been using the Netty implemented classes like [SupportedCipherSuiteFilter](https://github.com/netty/netty/blob/4.1/handler/src/main/java/io/netty/handler/ssl/SupportedCipherSuiteFilter.java) for what I did and they all seemed to have that check, but you're absolutely right. It shouldn't be needed. 

I'll look at the validation of how these objects will be generated

Reply:
well, that's ominous

Reply:
If I'm reading https://github.com/netty/netty/commit/04f77b76f802a5bb30447eb0e3580f6ba0e1f2db correctly the Netty impl validates earlier in the process so the break on `null` should never actually happen. That said I think @robobario is right we should continue rather than break.